### PR TITLE
Export proj target to scope visible for test/unit target

### PR DESCRIPTION
**Pull request by mloskot**
Refines #1034 

----

Addresses @rouault comment https://github.com/OSGeo/proj.4/commit/e9b399fc91e72b3cf81337e6b4dca8bc60fafaf8#r29288219

### CODE REVIEW

File: test/unit/basic_test.cpp [hunk 1]
```diff
@@ -27,13 +27,15 @@
...
+#include <proj.h>
 
 namespace {
 
 // ---------------------------------------------------------------------------
 
 TEST( dumy, dummy ) {
-    ASSERT_EQ( 1 , 1 );
+    PJ* p = ::proj_create(PJ_DEFAULT_CTX, "+proj=merc +ellps=clrk66 +lat_ts=33");
+    ASSERT_NE(p, nullptr);
```
Comment (rouault) [inline test/unit/basic_test.cpp]: you need a ```::proj_destroy(p);``` to make leak sanitizer happy :-)

Comment (mloskot) [reply test/unit/basic_test.cpp]: I blame the `dummy` name :-)
It should be fixed now.
For serious tests, that's gonna be taken care of by fixture(s).

**Comment by rouault** [comment]
I've merged your branch with a needed extra commit to set the PATH for the tests on Windows:  https://github.com/OSGeo/proj.4/commit/e37ad143d71f8d1b76218c30bcc8612d12724630

**Comment by mloskot** [comment]
@rouault I see. By default, with VS, static libraries are built for both, gtest.lib and proj_5_0.lib, so I didn't notice the issue. Looks like your tweak may be required due to `-DBUILD_LIBPROJ_SHARED=ON` in `appveyor.yml`. Anyhow, thanks!

**Pull request merged**